### ISSUE TITLE
Xcode 9 Main Thread Sanitizer

### ISSUE THREAD

**Issue by eneko** [Instabug/Instabug-iOS#163]
Seems like the Instabug SDK is trying to update the network indicator from a background thread.

Happens both in versions 6.4.1 and 7.2.7

```
Main Thread Checker: UI API called on a background thread: -[UIApplication setNetworkActivityIndicatorVisible:]
PID: 6247, TID: 25758465, Thread name: (none), Queue name: NSOperationQueue 0x610000220920 :: NSOperation 0x60000025bf30 (QOS: DEFAULT), QoS: 21
Backtrace:
4   Instabug                            0x000000011173a31a -[IBGServerManager updateNetworkActivityIndicator] + 146
5   Instabug                            0x000000011173a271 -[IBGServerManager refresh] + 301
6   Instabug                            0x0000000111737250 -[IBGSurveyManager getSurveys] + 100
7   Instabug                            0x0000000111737bdd -[IBGSurveyManager observeValueForKeyPath:ofObject:change:context:] + 120
8   Foundation                          0x0000000115ddc089 NSKeyValueNotifyObserver + 349
9   Foundation                          0x0000000115ddb93e NSKeyValueDidChange + 483
10  Foundation                          0x0000000115ebb905 NSKeyValueDidChangeWithPerThreadPendingNotifications + 140
11  Instabug                            0x0000000111722417 -[IBGFeatures setFeatureOfName:toValue:isUserSet:] + 242
12  Instabug                            0x0000000111721f20 -[IBGFeatures JSONSetFeature:toValue:] + 112
13  Instabug                            0x000000011172220f -[IBGFeatures featuresRequestDone:isSuccess:response:error:] + 714
14  Instabug                            0x00000001116f7c7e -[IBGFeaturesRequest dataServerExecutionDone:statusCode:error:response:] + 285
15  Instabug                            0x00000001116f7798 __75+[IBGServerDataRequest executeWithURL:HTTPMethod:payload:headers:delegate:]_block_invoke_2 + 111
16  Instabug                            0x00000001116f7410 -[IBGServerDataRequest prepareResponseWithData:error:rawResponse:completionHandler:] + 517
17  Instabug                            0x00000001116f76fa __75+[IBGServerDataRequest executeWithURL:HTTPMethod:payload:headers:delegate:]_block_invoke + 153
18  xxxx                                0x000000010e3ba539 __65+[NRMANSURLConnectionSupport poseImplementationBlockForSelector:]_block_invoke_3 + 585
19  CFNetwork                           0x00000001194f577f __67+[NSURLConnection sendAsynchronousRequest:queue:completionHandler:]_block_invoke_2 + 161
20  Foundation                          0x0000000115ddce7b __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
21  Foundation                          0x0000000115ddca46 -[NSBlockOperation main] + 92
22  Foundation                          0x0000000115ddb098 -[__NSOperationInternal _start:] + 628
23  Foundation                          0x0000000115dd7170 __NSOQSchedule_f + 198
24  libdispatch.dylib                   0x0000000118be56c8 _dispatch_client_callout + 8
25  libdispatch.dylib                   0x0000000118bee217 _dispatch_queue_serial_drain + 1272
26  libdispatch.dylib                   0x0000000118beec6c _dispatch_queue_invoke + 342
27  libdispatch.dylib                   0x0000000118bf1ef4 _dispatch_root_queue_drain + 783
28  libdispatch.dylib                   0x0000000118bf1b82 _dispatch_worker_thread4 + 54
29  libsystem_pthread.dylib             0x00000001190385a2 _pthread_wqthread + 1299
30  libsystem_pthread.dylib             0x000000011903807d start_wqthread + 13
```

**Comment by JackoPlane** [comment]
+1

**Comment by Figo0o** [comment]
Hello @eneko, Thanks for reporting this out. Currently we are working to full support iOS 11. We should have a release ASAP, Let me notify you then.

**Comment by Figo0o** [comment]
@eneko, @JackoPlane  

We have just released V7.2.8 that should solves this problem and other issues appeared with iOS 11. Please let me know if I can help with anything else. 

Thanks!

**Comment by TinyQ** [comment]
看来不能还原这个版本了：https://fabric.io/zhihu/ios/apps/com.zhihu.ios/issues/5ae5275f638393737a9b1ffc?time=last-thirty-days
